runtime/compiler/control/CompilationThread.cpp

```diff
@@ -3760,6 +3764,34 @@ TR::CompilationInfoPerThread::initializeSegmentCache(J9::J9SegmentProvider &segm
    return segmentCache;
    }
 
+bool
+TR::CompilationInfo::checkVersion(JITaaS::J9ClientStream *client)
+   {
+   uint32_t clientMajorNumber = MAJOR_NUMBER;
+   uint32_t ClientMinorNumber = MINOR_NUMBER;
+   uint32_t ClientPatchNumber = PATCH_NUMBER;
+   client->write(clientMajorNumber , ClientMinorNumber, ClientPatchNumber);
+   client->read(); //read and discard the J9ServerMessageType::compilationCode.
+   return std::get<0>(client->getRecvData<bool>()); //Read the version check result.
+   }
+
+bool
+TR::CompilationInfo::checkVersion(JITaaS::J9ServerStream *server)
+   {
+   auto recv = (server->read<uint32_t, uint32_t, uint32_t>());
+   bool result = false;
+   uint32_t clientMajorNumber = std::get<0>(recv);
```

Comment:
I think we can do with just one uint32_t. The patch and the major version are not going to change too often, so we can use only 8 bits, while for the minor version we can use 16 bits.

Or, since the minor version never causes an incompatibility check, we could skip sending it. Thus we could use uint16_t for all 3 version numbers, but send only two of them (the major and the minor)